tests/python_compiler/transforms/test_xdsl_measurements_from_samples.py

```diff
@@ -0,0 +1,876 @@
+                qml.expval,
+                qml.Y,
+                -1.0,
+                marks=pytest.mark.xfail(
+                    reason="Only PauliZ-basis measurements supported",
+                    strict=True,
+                    raises=NotImplementedError,
+                ),
+            ),
+            pytest.param(
+                partial(qml.RX, phi=-np.pi / 2),
+                qml.var,
+                qml.Y,
+                0.0,
+                marks=pytest.mark.xfail(
+                    reason="Only PauliZ-basis measurements supported",
+                    strict=True,
+                    raises=NotImplementedError,
+                ),
+            ),
+            pytest.param(
+                partial(qml.RX, phi=np.pi / 2),
+                qml.var,
+                qml.Y,
+                0.0,
+                marks=pytest.mark.xfail(
+                    reason="Only PauliZ-basis measurements supported",
+                    strict=True,
+                    raises=NotImplementedError,
+                ),
+            ),
+        ],
+    )
+    # pylint: disable=too-many-arguments,too-many-positional-arguments
+    def test_exec_1_wire_mp_with_obs(self, shots, initial_op, mp, obs, expected_res):
+        """Test the measurements_from_samples transform on a device with a single wire and terminal
+        measurements that require an observable (i.e. expval and var).
+        """
+
+        dev = qml.device("lightning.qubit", wires=1, shots=shots)
+
+        @qml.qnode(dev)
+        def circuit_ref():
+            initial_op(wires=0)
+            return mp(obs(wires=0))
+
+        assert expected_res == circuit_ref(), "Sanity check failed, is expected_res correct?"
+
+        circuit_compiled = qml.qjit(
+            measurements_from_samples_pass(circuit_ref),
+            pass_plugins=[xdsl_plugin.getXDSLPluginAbsolutePath()],
+        )
+
+        assert expected_res == circuit_compiled()
+
+    # -------------------------------------------------------------------------------------------- #
+
+    @pytest.mark.usefixtures("enable_disable_plxpr")
+    @pytest.mark.parametrize("shots", [1, 2])
+    @pytest.mark.parametrize(
+        "initial_op, expected_res",
+        [
+            (qml.I, [1.0, 0.0]),
+            (qml.X, [0.0, 1.0]),
+        ],
+    )
+    def test_exec_1_wire_probs(self, shots, initial_op, expected_res):
+        """Test the measurements_from_samples transform on a device with a single wire and terminal
+        probs measurements.
+        """
+
+        dev = qml.device("lightning.qubit", wires=1, shots=shots)
+
+        @qml.qnode(dev)
+        def circuit_ref():
+            initial_op(wires=0)
+            return qml.probs(wires=0)
+
+        assert np.array_equal(
+            expected_res, circuit_ref()
+        ), "Sanity check failed, is expected_res correct?"
+
+        circuit_compiled = qml.qjit(
+            measurements_from_samples_pass(circuit_ref),
```

Comment:
Hey @joeycarter , just wondering why we donot need to pass a qfunc to `qjit` here?